.github/workflows/integration-test.yaml

```diff
@@ -0,0 +1,103 @@
+name: Integration Tests
+
+#required variables
+#TESTIM_PROJECT
+#TESTIM_USER
+
+#Reqiured secrets
+#TESTIM_TOKEN
+#OIDC_URL
+#OIDC_CLIENT
+#OIDC_CLIENT_SECRET
+#AUTH_API_URL
+#PAY_API_URL
+#TEST_USER
+#TEST_PASSWORD
+#CANADA_POST_API_KEY
+#AUTH_URL
+#AUTH_REALM
+#REGISTRY_HOME_URL
+#NUXT_CLIENT_ID
+
+#May be needed in future
+#FIREBASE_API_KEY
+#FIREBASE_MESSAGING_SENDER_ID
+#FIREBASE_APP_ID
+
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  integration_tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+      - name: terraform
+        env:
+          TERRAFORM_VERSION: "1.8.0"
+        run: |
+          export WD=`pwd`
+          cd /tmp
+          wget -q https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+          unzip -o /tmp/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+          rm -rf /tmp/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+          chmod +x terraform
+          alias terraform='/tmp/terraform'
+          cd $WD/api
+          docker build . -t regapi:latest
+          cd $WD/web/site
+          docker build . -t regweb:latest
+          cd $WD/terraform
+          mkdir _tmp
+          chmod -R 777 _tmp
+          
+          echo "127.0.0.1 ar.example.demo" | sudo tee -a /etc/hosts
+          echo "127.0.0.1 ar.example.demo" | sudo tee -a /etc/hosts
+          terraform init
+
+          #create terraform vars file
```

Comment:
For this part consider using the 1password Vault service or google secret manager to retrieve the value. And don't need to put so many values in the github.